File: app/src/main/java/co/edu/uniautonoma/posgradosapp/Actividades/EscuelaActivity.java
package co.edu.uniautonoma.posgradosapp.Actividades;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;

import co.edu.uniautonoma.posgradosapp.R;
import co.edu.uniautonoma.posgradosapp.ViewModels.EscuelaViewModel;
import xdroid.toaster.Toaster;

public class EscuelaActivity extends BaseActivity {

    EscuelaViewModel escuelaViewModel;
    TextView tvDescripcionescuela;
    TextView tvDirector;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_escuela);

        tvDescripcionescuela = (TextView) findViewById(R.id.tvDescripcionescuela);
        tvDirector = (TextView) findViewById(R.id.tvDirector);

        ObtenerEscuela();
    }

    private void ObtenerEscuela(){

        mostrarDialog();
        escuelaViewModel = ViewModelProviders.of(this).get(EscuelaViewModel.class);

        if(escuelaViewModel.getEscuela()!=null) {
            escuelaViewModel.getEscuela().observe(this, escuela -> {
                tvDescripcionescuela.setText(escuela.getDescripcion());
                String director = "Director: " + escuela.getDirector();
                tvDirector.setText(director);
            });
        }else{
            ocultarDialog();
            Toaster.toast(R.string.EstadoServidor);
        }
        ocultarDialog();
    }
}

File: presentation/src/main/java/co/edu/uniautonoma/posgradosapp/presentation/ui/escuela/EscuelaActivity.kt
package co.edu.uniautonoma.posgradosapp.presentation.ui.escuela

import android.os.Bundle
import android.util.Log
import androidx.lifecycle.Observer
import co.edu.uniautonoma.posgradosapp.presentation.R
import co.edu.uniautonoma.posgradosapp.domain.entity.Escuela
import co.edu.uniautonoma.posgradosapp.presentation.ui.base.BaseActivity
import kotlinx.android.synthetic.main.activity_escuela.*
import org.koin.androidx.viewmodel.ext.android.viewModel
import xdroid.toaster.Toaster

class EscuelaActivity : BaseActivity() {

    private val escuelaViewModel: EscuelaViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_escuela)

        mostrarDialog()
        observarResultados()
    }

    private fun observarResultados() {
        escuelaViewModel.escuelaLiveData.observe(this, Observer{
            setViews(it)
            ocultarDialog()
        })

        escuelaViewModel.errorLiveData.observe(this, Observer {
            ocultarDialog()
            Toaster.toast(R.string.EstadoServidor)
            Log.d("EscuelaError:",  it)
        })
    }

    private fun setViews(it: Escuela) {
        val descripcion = it.descripcion
        val director = "Director: ${it.director}"

        tvDescripcionescuela.text = descripcion
        tvDirector.text = director
    }
}